### PR TITLE
chore: create and configure repo for testing

### LINKED ISSUE #1: Add and configure repo to run tests
**Description**
We need to set up the repository to allow tests to be run against a local fork. The following things need to be done:
- Typescript should be added and configured
- esLint and prettier should be added and configured
- Hardhat should be added and configured
- [Mento Core Ts](https://github.com/mento-protocol/mento-core-ts)  should be installed to give access to governance contracts

Mento Core TS does not contain references to the addresses of the deployed contracts. We will need to also reference the SDK to get the latest addresses of the contracts.

In addition to the above, a node script should be added to do the following in one command (e.g. test:alfajores):
- Start a local fork of a specified network 
- Run the tests using hardhat

Once the above has been done, a sample test (e.g. Get the total supply of mento token) should be added to verify the repo was set up correctly. 

**Acceptance Criteria**
> Generated by [Zenhub AI](https://www.zenhub.com/ai)

- [x] **Scenario 1: Add and configure Typescript**
  - The repository should have Typescript added and configured.
  - The code should be able to compile successfully using Typescript.

- [x] **Scenario 2: Add and configure esLint and prettier**
  - The repository should have esLint and prettier added and configured.
  - The code should follow the defined linting and formatting rules.

- [ ] **Scenario 3: Add and configure Jest**
  - Jest should be added to the repository.
  - Basic configuration for Jest should be set up.
  - The tests should be able to run successfully using Jest.

- [x] **Scenario 4: Add and configure Hardhat**
  - Hardhat should be added to the repository.
  - Hardhat should be configured properly.
  - The repository should be able to interact with the specified network using Hardhat.

- [x] **Scenario 5: Install Mento Core Ts**
  - Mento Core Ts should be installed in the repository.
  - The installation should provide access to governance contracts.

- [x] **Scenario 6: Add node script for local fork and test execution**
  - A node script should be added to the repository.
  - The script should be able to start a local fork of a specified network.
  - The script should be able to run tests using Jest.

- [x] **Scenario 7: Verify repository setup with a sample test**
  - A sample test should be added to the repository.
  - The test should verify the correct setup of the repository.
  - The test should check the total supply of the mento token.

### PR DESCRIPTION
**Pull request by bayological**
### Description

Setup and configured dependencies & added two tests to verify setup. 

### Related issues

- Fixes #1

### CODE REVIEW

**Review by baroooo — COMMENTED** [review]
I don't think we need foundry in this repo. We should be able to run tests using only hardhat. Compiling contracts also seemed unnecessary since we only need types which can be fetched from mento-core-ts repo. I will give it a go, let me know what you think

**Comment by bayological** [comment]
> I don't think we need foundry in this repo. We should be able to run tests using only hardhat. Compiling contracts also seemed unnecessary since we only need types which can be fetched from mento-core-ts repo. I will give it a go, let me know what you think

Yes, you're right, we definitely don't. Let me know how you get on.

File: package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
   "scripts": {
     "test": "NETWORK=celo pnpm exec hardhat test",
     "test:alfajores": "NETWORK=alfajores pnpm exec hardhat test",
-    "test:baklava": "NETWORK=baklava pnpm exec hardhat test",
+    "test:baklava": "NETWORK=baklava pnpm exec hardha test",
+    "build": "tsc",
```
Comment (bayological) [inline package.json:10]: We should change this to `pnpm tsc` so we use the projects Typescription version instead of the global one.

**Review by bayological — COMMENTED** [review]
LGTM but can't approve as I opened this.

**Review by nvtaveras — APPROVED** [review]

**Pull request merged**